core/src/main/java/hudson/Util.java

```diff
@@ -331,10 +331,20 @@ public static void deleteRecursive(@Nonnull File dir) throws IOException {
      */
     //Taken from http://svn.apache.org/viewvc/maven/shared/trunk/file-management/src/main/java/org/apache/maven/shared/model/fileset/util/FileSetManager.java?view=markup
     public static boolean isSymlink(@Nonnull File file) throws IOException {
-        Boolean r = isSymlinkJava7(file);
-        if (r != null) {
-            return r;
-        }
+        /*
+         *  Windows Directory Junctions are effectively the same as Linux symlinks to directories.
+         *  Unfortunately, the Java 7 NIO2 API function isSymbolicLink does not treat them as such.
+         *  It thinks of them as normal directories.  To use the NIO2 API & treat it like a symlink,
+         *  you have to go through BasicFileAttributes and do the following check:
+         *     isSymbolicLink() || isOther()
+         *  The isOther() call will include Windows reparse points, of which a directory junction is.
+         *
+         *  Since we already have a function that detects Windows junctions or symlinks and treats them
+         *  both as symlinks, let's use that function and always call it before calling down to the
+         *  NIO2 API.
+         *
+         *  This is tracked as JENKINS-29956.
```

Comment:
@daniel-beck I respectfully disagree.  The reason the code is like this is due to JENKINS-29956.  The comment should direct devs to a place where they can find out more information and `git blame` via the github UI does not support ignoring whitespace changes ad as such can not be relied upon to link these changes to the actual reason for the change in the future.

Now if you want to request the link is handled differently - "for the rationale here see JENKINS-XXXX" but it is important to me that this link is left in the code.